tests/unit/test_charm_2.py

```diff
@@ -1,12 +1,48 @@
+import os
```

Comment:
Would it be possible to come up with a more descriptive name for the file than `test_charm_2`? E.g., instead of `2` maybe something about how it is different or what specifically it tests?

Reply:
The 2 here is because I don't want to change the original `test_charm.py` since it has some conflicts between the old code and new code. I will remove the `test_charm.py` and rename `test_charm_2.py` to `test_charm.py` in final stage.